### PR TITLE
Load C++ library on executors

### PR DESCRIPTION
**Pull request by pratikdhandharia**
Adding a Wrapper API to load/unload C++ library on all executors. 

@afard @shreya2k7 Can you guys take a look ?

### CODE REVIEW

File: platform/master/R/master.R [hunk 1]
```diff
@@ -443,3 +443,37 @@ check_dr_version_compatibility<-function(){
...
 }
 
+dload <- function(x, trace = FALSE){
+
+  number_of_executors <- sum(distributedR_status()$Inst)
+  y <- lapply(x, dloadHelper, trace=trace, num_of_Exec = number_of_executors)
+
+}
+
+dloadHelper <- function(x, trace = FALSE, num_of_Exec) {
```
Comment (shreya2k7) [inline platform/master/R/master.R]: Pls make helper functions hidden and possible to change the name to be more specific to loading libraries?

**Comment by shreya2k7** [comment]
Loosk good otherwise

**Comment by afard** [comment]
1) Hiding the helper functions is a good point.
2) Regarding the name of the functions, we should note that they are distributed version of "dyn.load()" and "dyn.unload() functions. In R terminology, a "library" function is loading a package not an ".so" file. I am not sure about any better name. The other options might be "dloadSO" and "dunloadSO", or "d.dyn.load" and "d.dyn.unload"?

**Comment by pratikdhandharia** [comment]
We can ask others as well for the name of function. 

@jorgemarsal  @fun-indra @etduwx @vishrutg @shreya2k7 What do you guys think about the function name?

Also I was thinking of adding the man page for these functions. I think it would be good.

**Pull request merged**